### PR TITLE
fix(cli): prepare the  string of all model settings and use it in the…

### LINKED ISSUE #2522: [CLI] Adding nested configurations to model settings
## Description / Steps to reproduce / Feature proposal
We need to add some configuration to the model settings like

```ts
@model({ 
  settings: {
    mongodb: {
      collection: 'clients',
    },
  },
})
```

## Current Behavior
In configuration.JSON I put:

```ts
"modelSettings": {
			"mongodb": {
                                 "collection": "clients"
                          }
  },
```

The CLI generated model has a string instead of JSON:

```ts
@model({ settings: { "mongodb": { "collection": "clients" } } })
```

## Expected Behavior
The generated output should be 

```ts
@model({ settings: { 
                     mongodb: { 
                            collection: "clients"
            } 
    } })
```

I think in @loopback\cli\generators\model\templates\model.ts.ejs
instead of  `JSON.stringify(modelSettings)` should be `JSON.stringify(modelSettings).replace(/\"([^(\")"]+)\":/g,"$1:")`

### PR DESCRIPTION
**Pull request by yanamg7**
… template

prepare the  string of all model settings and use it in the template, fix one test and add the new

fix #2522

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/cli/generators/model/index.js [hunk 1]
```diff
@@ -215,6 +215,22 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
...
 
+        if (this.artifactInfo.modelSettings) {
+          let output = '';
+          for (var key in this.artifactInfo.modelSettings) {
+            if (output.length > 0) {
+              output += ',\n';
+            }
+            output +=
+              key +
+              ' : ' +
```
Comment (bajtos) [inline packages/cli/generators/model/index.js]:
```suggestion
              ': ' +
```

File: packages/cli/generators/model/index.js [hunk 2]
```diff
@@ -215,6 +215,22 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
...
+              JSON.stringify(this.artifactInfo.modelSettings[key], null, 5)
```
Comment (bajtos) [inline packages/cli/generators/model/index.js]: Why 5 characters for indentation? We use 2 characters everywhere else in LoopBack.

Also what happens when `this.artifactInfo.modelSettings[key]` is an object? I think `JSON.stringify` will return multiple lines of text. Only the first line will preserve indentation, the remaining lines will have too little indent.

```ts
@model({
  custom: {
foo: 'bar'
},
})
```

Let's use Yeoman's [stringify-object](https://www.npmjs.com/package/stringify-object) package please to handle the complex details for us.

```ts
this.artifactInfo.modelSettings = stringifyObject(this.artifactInfo.modelSettings, {
  indent: '  ', // two spaces
  singleQuotes: true,
  inlineCharacterLimit: 80,
});
```

File: packages/cli/generators/model/templates/model.ts.ejs [hunk 3]
```diff
@@ -1,35 +1,39 @@
 <% if (isModelBaseBuiltin) { -%>
-import {<%= modelBaseClass %>, model, property} from '@loopback/repository';
-<% } else { -%>
-import {model, property} from '@loopback/repository';
-import {<%= modelBaseClass %>} from '.';
-<% } -%>
+  import {<%= modelBaseClass %>, model, property} from '@loopback/repository';
```
Comment (bajtos) [inline packages/cli/generators/model/templates/model.ts.ejs]: Please revert all whitespace-only changes in this file.

Comment (yanamg7) [reply packages/cli/generators/model/templates/model.ts.ejs]: I didn't change this lines but I think that this is the not isModelBaseBuiltin case and modelBaseClass import should be from current folder, not from loopback/repository

File: packages/cli/generators/model/index.js (hunk 2 above)
Comment (yanamg7) [reply packages/cli/generators/model/index.js]: Very nice tool, thank you. Commited

File: packages/cli/generators/model/templates/model.ts.ejs (hunk 3 above)
Comment (yanamg7) [reply packages/cli/generators/model/templates/model.ts.ejs]: Sorry, I didn't understand. Reverted.

File: packages/cli/generators/model/index.js (hunk 2 above)
Comment (bajtos) [reply packages/cli/generators/model/index.js]: @yanamg7 I still see `JSON.stringify` being used. Did you perhaps forget to push your changes?

File: packages/cli/generators/model/templates/model.ts.ejs (hunk 3 above)
Comment (bajtos) [reply packages/cli/generators/model/templates/model.ts.ejs]: Ditto, I don't see these changes being reverted.

**Comment by bajtos** [comment]
> @yanamg7 closed this 2 days ago

Hey, what happened? I would love to see this pull request finished and landed. Do you need any help?

**Comment by bajtos** [comment]
@yanamg7 let me know how can I help you!

**Comment by yanamg7** [comment]
Hi,

I failed with git - revert of spaces failed, conflicts end e.t.c It took to match time for this minorly issue. I’m sorry

**Comment by bajtos** [comment]
> I failed with git - revert of spaces failed, conflicts end e.t.c

I see. In the future, please open a new feature branch for pull requests, it will make it much easier to make changes later. See https://loopback.io/doc/en/lb4/submitting_a_pr.html#2-create-the-feature-branch

> It took to match time for this minorly issue. I’m sorry

No worries. I'll try to cherry-pick your changes into a new pull request.

**Comment by bajtos** [comment]
Here is my follow-up pull request, I put you @yanamg7 as a co-author to preserve credits: https://github.com/strongloop/loopback-next/pull/2843

**Pull request closed without merging**